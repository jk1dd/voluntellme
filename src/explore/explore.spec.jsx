import React, { Component } from 'react';
import Explore from './explore'
import SearchBar from './searchBar'
import OrgTable from './orgTable'
import { shallow } from 'enzyme';

describe('<Explore />', () => {
  it('should not crash', () => {
    const explore = shallow(<Explore />)
    expect(explore).toBeTruthy
  })

  it('should render an h1', () => {
    const explore = shallow(<Explore />)
    expect(explore.find('h1').length).toEqual(1)
  })

  it('should render a header title', () => {
    const explore = shallow(<Explore />)
    expect(explore).toMatchSnapshot()
  })

  it('should render SearchBar', () => {
    const explore = shallow(<Explore />)
    expect(explore.find(SearchBar).length).toEqual(1)
  })

  it('should render OrgTable', () => {
    const explore = shallow(<Explore />)
    expect(explore.find(OrgTable).length).toEqual(1)
  })


})
