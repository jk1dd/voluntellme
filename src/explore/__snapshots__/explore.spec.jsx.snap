// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Explore /> should render a header title 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <div>
    <h1>
        Look at all these organizations!
    </h1>
    <SearchBar
        filterText=""
        onFilterTextInput={[Function]}
    />
    <OrgTable
        filterText=""
        orgs={Array []}
    />
</div>,
  "nodes": Array [
    <div>
      <h1>
            Look at all these organizations!
      </h1>
      <SearchBar
            filterText=""
            onFilterTextInput={[Function]}
      />
      <OrgTable
            filterText=""
            orgs={Array []}
      />
</div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <Explore />,
      "_debugID": 5,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": Explore {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "handleFilterTextInput": [Function],
        "props": Object {},
        "refs": Object {},
        "state": Object {
          "filterText": "",
          "organizations": Array [],
        },
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 3,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <div>
          <h1>
                    Look at all these organizations!
          </h1>
          <SearchBar
                    filterText=""
                    onFilterTextInput={[Function]}
          />
          <OrgTable
                    filterText=""
                    orgs={Array []}
          />
</div>,
        "_debugID": 6,
        "_renderedOutput": <div>
          <h1>
                    Look at all these organizations!
          </h1>
          <SearchBar
                    filterText=""
                    onFilterTextInput={[Function]}
          />
          <OrgTable
                    filterText=""
                    orgs={Array []}
          />
</div>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <Explore />,
}
`;
